e2e/e2e_tests.py

```diff
@@ -39,105 +46,119 @@ def _stop_containers():
-        }
-        r = client.post(login_url, data=login_data)
-        return {'request': r, 'client': client}
+    def test_create_new_excerpt_succeeds(self):
+        self._login()
+        self._create_new_order()
+        self._go_to_order()
+        links = self._wait_for_order_to_complete_and_fetch_download_links()
+        self.assertEqual(len(links), 8)
+        download_links = self._get_download_links_from_links(links)
+        for download_link in download_links:
+            self._download_and_test_zip_contents(download_link)
+
+    # Helper methods
+    def _login(self):
+        self.browser.get(self._make_link('/login/'))
+        login_form = self.browser.find_element_by_id('osmaxx-login-form')
+        username = self.browser.find_element_by_id('id_username')
+        password = self.browser.find_element_by_id("id_password")
+        username.send_keys('admin')
+        password.send_keys('admin')
+        login_form.find_element_by_xpath("//input[@type='submit']").click()
+
+    def _logout(self):
+        self.browser.get(self._make_link('/logout/'))
+
+    def _create_new_order(self):
+        self.browser.get(self._make_link('/orders/new/#new-excerpt'))
+        # wait for page to load
+        WebDriverWait(self.browser, 20).until(
+            expected_conditions.presence_of_element_located((By.ID, "new_excerpt_name"))
+        )
+        for html_id, send_value in self.excerpt_data['send_keys'].items():
+            element = self.browser.find_element_by_id(html_id)
+            element.clear()
+            element.send_keys(send_value)
+        for html_id in self.excerpt_data['click']:
+            self.browser.find_element_by_id(html_id).click()
+        self.browser.find_element_by_xpath("//input[@type='submit']").click()
+        self.browser.get(self._make_link('/orders/'))
+
+    def _go_to_order(self):
+        xpath_link_to_order = '/html/body/div/div/div[2]/div/h3/a'
+        WebDriverWait(self.browser, 60).until(
+            expected_conditions.presence_of_element_located((By.XPATH, xpath_link_to_order))
+        )
+        self.browser.find_element_by_xpath(xpath_link_to_order).click()
+
+    def _wait_for_order_to_complete_and_fetch_download_links(self):
+        poll_frequency_in_seconds = 10
+        waited_time = 0
+        status_element = self._get_status_element()
+        while '✓' not in status_element:
+            self.assertFalse('∅' in status_element)
+
+            waited_time += poll_frequency_in_seconds
+            sleep(poll_frequency_in_seconds)
+
+            if waited_time >= self.wait_time_seconds:
+                raise selenium_exceptions.TimeoutException
+            else:
+                self.browser.get(self.browser.current_url)
+                status_element = self._get_status_element()
+        links = self._get_download_a_tags()
+        return links
+
+    def _get_download_links_from_links(self, links):
+        return [links[i].get_attribute('href') for i in range(0, 7, 2)]
```

Comment:
Replace magic numbers with constants or expressions based on constants.
##### Suggestion
    NUMBER_OF_LINKS_PER_DOWNLOAD = 2
    EXPECTED_NUMBER_OF_DOWNLAODS = 4
    def _get_download_links_from_links(self, links):
        return [links[i].get_attribute('href') for i in range(0, (NUMBER_OF_LINKS_PER_DOWNLOAD * EXPECTED_NUMBER_OF_DOWNLAODS) - 1, NUMBER_OF_LINKS_PER_DOWNLOAD)]